public/app/plugins/panel/geomap/components/DataHoverRows.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import { Collapse, TabContent, useStyles2 } from '@grafana/ui';
+import { DataFrame, FieldType, getFieldDisplayName, GrafanaTheme2 } from '@grafana/data';
+import { css } from '@emotion/css';
+import { FeatureLike } from 'ol/Feature';
+
+import { GeomapLayerHover } from '../event';
+import { DataHoverRow } from './DataHoverRow';
+import { isString } from 'lodash';
+
+type Props = {
+  layers?: GeomapLayerHover[];
+  activeTabIndex: number;
+};
+
+export const DataHoverRows = ({ layers, activeTabIndex }: Props) => {
+  const styles = useStyles2(getStyles);
+  const [rowMap, setRowMap] = useState(new Map<string | number, boolean>());
+
+  const updateRowMap = (key: string | number, value: boolean) => {
+    setRowMap(new Map(rowMap.set(key, value)));
+  };
+
+  const rows =
+    (layers &&
+      layers.map(
+        (geomapLayer, index) =>
+          index === activeTabIndex && (
+            <div key={geomapLayer.layer.getName()}>
+              <div>
+                {geomapLayer.features.map((feature, idx) => {
+                  const key = feature.getId() ?? idx;
+                  const shouldDisplayCollapse = geomapLayer.features.length > 1;
+
+                  return shouldDisplayCollapse ? (
+                    <Collapse
+                      key={key}
+                      collapsible
+                      label={generateLabel(feature, idx)}
+                      isOpen={rowMap.get(key)}
+                      onToggle={() => {
+                        updateRowMap(key, !rowMap.get(key));
+                      }}
+                      className={styles.collapsibleRow}
+                    >
+                      <DataHoverRow feature={feature} />
+                    </Collapse>
+                  ) : (
+                    <DataHoverRow key={key} feature={feature} />
+                  );
+                })}
+              </div>
+            </div>
+          )
+      )) ??
+    null;
+
+  return <TabContent>{rows}</TabContent>;
```

Comment:
maybe inline?  we already check that layers exists
```suggestion
  return <TabContent>{layers.map( ...) }</TabContent>;
```